plugins/pencilblue/controllers/actions/admin/sites/delete_site.js

```diff
@@ -0,0 +1,50 @@
+/*
+ Copyright (C) 2015  PencilBlue, LLC
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+module.exports = function DeleteSiteActionModule(pb) {
+
+    //pb dependencies
+    var util = pb.util;
+
+    /**
+     * Deletes a current site
+     */
+    function DeleteSiteAction(){}
+    util.inherits(DeleteSiteAction, pb.BaseController);
+
+    DeleteSiteAction.prototype.render = function(cb) {
+        var self = this;
+        var siteid = self.pathVars.siteid;
+        var siteQueryService = new pb.SiteQueryService();
+        siteQueryService.getCollections(function(err, allCollections) {
+            siteQueryService.deleteSiteSpecificContent(allCollections, siteid, function(err, result) {
+                if(util.isError(err)) {
+                    cb({
+                        code: 400,
```

Comment:
Is it a bad request (400) or a server error (500)?  PB allows you to callback with an error that has the message and ```code``` property set.  It will handle the formatting and sending of the specified code:

Some service:
```
Service.prototype.doSomething(cb) {
 var failures = BaseObjectService.validationFailure('someField', 'someField is required');
 var err = BaseObjectService.validationError([failures]);
 cb(err);
};
```

Controller (inside of handler):
```
this.service.doSomething(function(err, result) {
  if (util.isError(err)) {
    return cb(err);
  }
});
```

Server Output: HTTP Status - 400
```
{
  "message": "Validation Error",
  "validationErrors": [
    {
      "field": "someField",
      "message": "someField is required"
    }
  ]
}
```